Comparison Engine/Comparison Engine/Base Classes/BarManager.cs

```diff
@@ -16,26 +17,26 @@ private BarManager()
         }
 
         //A list of bars
-        public List<Bar> barList = new List<Bar>();
+        public Dictionary<int, Bar> barDictionary  { get; set; }
 
         //Adds a new bar
         public void AddBar(string barName, string location)
         {
-            int barID = barList.Count - 1;
-            barList.Add(new Bar(barID, barName, location));
+            int barID = barDictionary.Count - 1;
+            barDictionary.Add(barID, new Bar(barID, barName, location));
         }
 
         //Removes the bar from the list of all bars
         public void RemoveBar(Bar bar)
         {
-            barList.Remove(bar);
+            barDictionary.Remove(bar.barID);
         }
 
         //Finds the price of a specific drink at a specific bar
         public float FindDrinkPrice(int barID, int drinkID)
         {
             //Return the price of specified drink at this bar
-            if (barList[barID].availableDrinks.TryGetValue(drinkID, out float value))
+            if (barDictionary[barID].availableDrinks.TryGetValue(drinkID, out float value))
             {
                 return value;
             }
```

Comment:
Why do we try to get value, is there a scenario where we would allow a bar to have a drink without a specified price?

Reply:
This line was written by Kasparas. Can't really comment on that.

Reply:
will make a note to ask kasparas about this